design/fast_finality.md

```diff
@@ -0,0 +1,273 @@
+new light-batch chains and inspecting the results. To address this, an enclave's light-batch chain can only be 
+overwritten immediately after start-up. The recovered leader will poll its former followers for conflicting light 
+batches, and overwrite any light-batches at the same height. Once the leader starts normal execution, this overwriting 
+mechanism will be disabled (and thus an enclave restart, with the attendant start-up delay, must be incurred to 
+overwrite again and inspect the results).
+
+#### Alternatives considered
+
+This approach was selected over a number of alternatives.
+
+##### As above, but with `n` hosts all speaking to `m` enclaves
+
+The selected approach is simpler and more closely aligned to our current implementation (which assumes one enclave per 
+host, and vice-versa).
+
+##### Having a single node that is restored from backup
+
+This approach has several downsides:
+
+* Recovery would be much slower in this approach, as a governance action would be required to whitelist the new 
+  sequencer attestation in the management contract (this could be mitigated by preallocating some future enclave 
+  machines to use for failover, and whitelisting attestations for them in the management contract)
+* The latest light batches (those not contained in the backup) would have to be recovered from network peers, which 
+  would be more complicated than recovering them from specific, sequencer-operator controlled nodes. In particular, 
+  we'd to handle the case of a node receiving a light batch from the crashed leader that they then fail to gossip out 
+  correctly; some mechanism would have to be provided to allow them to return to the non-forked light-batch chain (e.g. 
+  overwriting the light-batch chain if they receive another with greater height)
+* This approach is more difficult operationally (creation, storage and recreation from backups)
+
+## Future work
+
+* Allowing nodes to challenge the sequencer (e.g. if the light batches are missing transactions, if a certain light 
+  batch is not included in the rollup, or if light batches and/or rollups are not produced at the correct frequency)
+* Creation of an inbox to allow transactions to be "forced through" if the sequencer is excluding them (the rough idea 
+  is that there will be an inbox for transactions in the management contract, and validators will reject light batches 
+  that do not contain any (valid) transactions that have sat in the inbox for a certain amount of time)
+
+## Unresolved issues
+
+* Select a design for preventing value extraction (see the section 
+  `Possible designs for preventing value-extraction`, below)
+* Do the light batches need to be linked to the latest block that was fed into the enclave?
+* How do we prevent denial-of-service attacks on the sequencer?
+* How do we achieve the desired cadence of light batches without tying ourselves to the L1 block cadence, in light of 
```

Comment:
I'd say this is part of the incentives mechanism, and out of scope in the phase 1, given the sequencer wants to ensure a good UX